gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/rest/api/service/sanitizer/HtmlSanitizerTest.java

```diff
@@ -121,6 +129,14 @@ private String getNotSafe() {
         return html;
     }
 
+    private String getXssNotSafe() {
+        String html = "";
+        html += "<!--->";
```

Comment:
I'm not sure I fully understand the original reasons why the XSS is not detected when there is an empty comment `<!--->` before the `<img src=x onerror=alert(1) />` but do you think it would be useful to add other test cases to show that we covered all the possibilities, ex:
one line
`<!----><img src=x onerror=alert(1) />`

2 lines
```
<!---->
<img src=x onerror=alert(1) />
```

several empty lines 
<!---->


<img src=x onerror=alert(1) />
```

Invalid comment
<!------->
<img src=x onerror=alert(1) />
```

Or maybe it is useless and has no consequences on how the next tag is interpreted?
Anyway, I'll leave it to you and will approve the PR.

Reply:
ok thanks! I added parametrised tests though to cover all scenarios